index.js

```diff
@@ -125,18 +125,44 @@ function Linux(instance, end) {
       }
       command.push(instance.command);
       command = command.join(' ');
-      Node.child.exec(command, { maxBuffer: MAX_BUFFER },
+      var cp = Node.child.exec(command, { maxBuffer: MAX_BUFFER },
         function(error, stdout, stderr) {
-          if (error) {
-            if (/No authentication agent found/.test(stderr)) {
-              error = new Error(NO_POLKIT_AGENT);
-            } else if (/Request dismissed|Command failed/i.test(error)) {
-              error = new Error(PERMISSION_DENIED);
+          if (!instance.options.stream) {
+            if (error) {
+              if (/No authentication agent found/.test(stderr)) {
+                error = new Error(NO_POLKIT_AGENT);
+              } else if (/Request dismissed|Command failed/i.test(error)) {
+                error = new Error(PERMISSION_DENIED);
+              }
             }
+            end(error, stdout, stderr);
           }
-          end(error, stdout, stderr);
         }
       );
+      if (instance.options.stream) {
+        var sentStreams = false;
+        var errored = false;
+        cp.stdout.once('data',
+          function(chunk) {
+            if (errored) return;
+            sentStreams = true;
+            end(null, cp.stdout, cp.stderr);
+          }
+        );
+        cp.stderr.on('data',
+          function(chunk) {
+            if (/No authentication agent found/.test(chunk)) {
```

Comment:
It's possible that "No authetication agen..." and other errors may be split across chunks.